app/controllers/histories_controller.rb

```diff
@@ -8,9 +8,45 @@ class HistoriesController < ApplicationController
   def create
     redirect_to root_url unless current_user.can_create_subject_history?
 
-    History.create!(patient_id: params.permit(:patient_id)[:patient_id],
+    history = History.new(patient_id: params.permit(:patient_id)[:patient_id],
+                          created_by: current_user.email,
+                          comment: params.permit(:comment)[:comment],
+                          history_type: params.permit(:type)[:type] || 'Comment')
+
+    history.original_comment = history
+
+    # Attempt to save and continue; else if failed redirect to index
+    render(json: history.errors, status: 422) && return unless history.save
+
+    render(json: history) && return
+  end
+
+  # "Edits" a history comment - a new history comment is created with the updated comment text and a reference to the id of the original
+  def edit
+    redirect_to root_url unless current_user.can_create_subject_history?
+
+    patient = current_user.viewable_patients.find_by(id: params.permit(:patient_id)[:patient_id])
+    history = patient.histories.find_by(id: params.permit(:id)[:id])
```

Comment:
need to add a `nil` check for patients before trying to access `patient.histories`

Reply:
added here: https://github.com/SaraAlert/SaraAlert/pull/793/commits/83e2b66ed498eb88bf55283e6ec1b25c43a1b2b9